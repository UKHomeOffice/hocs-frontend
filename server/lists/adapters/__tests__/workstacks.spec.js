const { dashboardAdapter, userAdapter, teamAdapter, workflowAdapter, stageAdapter } = require('../workstacks');
const { getUtcDateString } = require('../../../libs/dateHelpers');

const mockUser = { uuid: 1 };

const mockFromStaticList = jest.fn((list) => {
    switch (list) {
        case 'S_TEAMS':
            return 'MOCK_TEAM';
        case 'S_CASETYPES':
            return 'MOCK_CASETYPE';
        case 'S_STAGETYPES':
            return 'MOCK_STAGETYPE';
        case 'S_USERS':
            return 'MOCK_USER';
        default:
            return null;
    }
});

Date.now = jest.fn(() => 1609372800000); // 2020-12-31 00:00:00

const todaysDate = getUtcDateString(new Date(Date.now()));

const mockConfiguration = {
    workstackTypeColumns: [
        { workstackType: 'DEFAULT', workstackColumns: [] },
        { workstackType: 'WCS', workstackColumns: [] }
    ],
    deadlinesEnabled: true
};

const mockLogger = {
    debug: () => {
    },
    info: () => {
    },
    warn: () => {
    },
    error: () => {
    }
};

describe('Dashboard Adapter', () => {
    it('should transform a stage array to a dashboard schema', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '1900-01-01'
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-02'
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-03'
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '2200-04-01'
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: todaysDate
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-01'
                }
            ]
        };

        const result = await dashboardAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            configuration: mockConfiguration
        });
        expect(result).toMatchSnapshot();
    });

    it('should transform a stage array to a dashboard schema when deadlines are disabled', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '1900-01-01'
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-02'
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-03'
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '2200-04-01'
                }
            ]
        };
        const testConfiguration = {
            workstackTypeColumns: [
                { workstackType: 'DEFAULT', workstackColumns: [] },
                { workstackType: 'WCS', workstackColumns: [] }
            ],
            deadlinesEnabled: false
        }
            ;
        const result = await dashboardAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            configuration: testConfiguration
        });
        expect(result).toMatchSnapshot();
    });
});

describe('User Workstack Adapter', () => {
    it('should transform a stage array to a user workstack schema', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-01',
                    active: true
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-02',
                    active: true
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '2200-01-03',
                    active: false
                }
            ]
        };

        const result = await userAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            configuration: {
                workstackTypeColumns: [
                    { workstackType: 'DEFAULT', workstackColumns: [] },
                    { workstackType: 'WCS', workstackColumns: [] }
                ],
            }
        });
        expect(result).toMatchSnapshot();
    });
});

describe('Team Workstack Adapter', () => {
    it('should transform a stage array to a team workstack schema', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-01',
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-02',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-03',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-04',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-05',
                    caseReference: 'A/1234570/19',
                    active: true,
                    data: {}

                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '06-01-2200',
                    caseReference: 'A/1234569/19',
                    active: false,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '1900-01-01',
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: todaysDate,
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: todaysDate,
                    caseReference: 'A/1234567/19',
                    active: false,
                    data: {}
                },
            ]
        };

        const result = await teamAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            teamId: 2,
            configuration: {
                workstackTypeColumns: [
                    { workstackType: 'DEFAULT', workstackColumns: [] },
                    { workstackType: 'WCS', workstackColumns: [] }
                ], deadlinesEnabled: true
            }
        });

        expect(result).toMatchSnapshot();
    });

    it('should transform a stage array to a team workstack schema ordered by systemCalculatedPriority', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-01',
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: { systemCalculatedPriority: 6.05 }
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-02',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: { systemCalculatedPriority: 35.0 }
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-03',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: { systemCalculatedPriority: 13.43 }
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-04',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: { systemCalculatedPriority: 97.0 }
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-05',
                    caseReference: 'A/1234570/19',
                    active: true,
                    data: { systemCalculatedPriority: 43.9 }

                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '06-01-2200',
                    caseReference: 'A/1234569/19',
                    active: false,
                    data: { systemCalculatedPriority: 56.3 }
                },
            ]
        };

        const result = await teamAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            teamId: 2,
            configuration: {
                workstackTypeColumns: [
                    { workstackType: 'DEFAULT', workstackColumns: [] },
                    { workstackType: 'WCS', workstackColumns: [] }
                ], deadlinesEnabled: true
            }
        })
            ;
        expect(result).toMatchSnapshot();
    });
});

describe('Workflow Workstack Adapter', () => {
    it('should transform a stage array to a workflow workstack schema', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-01',
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: todaysDate,
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '1900-01-01',
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '1900-01-01',
                    caseReference: 'A/1234567/19',
                    active: false,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-02',
                    caseReference: 'A/1234568/19',
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-03',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-04',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-05',
                    caseReference: 'A/1234570/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-06',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'B',
                    userUUID: null,
                    deadline: '1900-01-07',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-08',
                    caseReference: 'A/1234569/19',
                    active: false,
                    data: {}
                }
            ]
        };

        const result = await workflowAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            teamId: 2,
            workflowId: 'WCS',
            configuration: mockConfiguration
        });
        expect(result).toMatchSnapshot();
    });
});

describe('Workflow Workstack Adapter', () => {
    it('should transform a stage array to a workflow workstack schema', async () => {
        const mockData = {
            stages: [
                {
                    teamUUID: 1,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-01',
                    caseReference: 'A/1234567/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-02',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 2,
                    deadline: '2200-01-03',
                    caseReference: 'A/1234568/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-04',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: null,
                    deadline: '1900-01-05',
                    caseReference: 'A/1234570/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-06',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'WCS',
                    stageType: 'B',
                    userUUID: null,
                    deadline: '1900-01-07',
                    caseReference: 'A/1234569/19',
                    active: false,
                    data: {}
                },
                {
                    teamUUID: 2,
                    caseType: 'DEFAULT',
                    stageType: 'A',
                    userUUID: 1,
                    deadline: '2200-01-08',
                    caseReference: 'A/1234569/19',
                    active: true,
                    data: {}
                }
            ]
        };

        const result = await stageAdapter(mockData, {
            user: mockUser,
            fromStaticList: mockFromStaticList,
            logger: mockLogger,
            teamId: 2,
            workflowId: 'WCS',
            stageId: 'A',
            configuration: mockConfiguration
        });
        expect(result).toMatchSnapshot();
    });
});