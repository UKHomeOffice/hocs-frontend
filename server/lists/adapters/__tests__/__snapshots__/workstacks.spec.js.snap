// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Adapter should transform a stage array to a dashboard schema 1`] = `
Object {
  "teams": Array [
    Card {
      "count": 3,
      "label": "MOCK_TEAM",
      "tags": Object {
        "allocated": 1,
        "overdue": 2,
      },
      "type": "team",
      "value": 1,
    },
    Card {
      "count": 3,
      "label": "MOCK_TEAM",
      "tags": Object {
        "allocated": 3,
        "overdue": 1,
      },
      "type": "team",
      "value": 2,
    },
  ],
  "user": Array [
    Card {
      "count": 1,
      "label": "Cases",
      "tags": Object {
        "allocated": undefined,
        "overdue": 1,
      },
      "type": undefined,
      "value": undefined,
    },
  ],
}
`;

exports[`Dashboard Adapter should transform a stage array to a dashboard schema when deadlines are disabled 1`] = `
Object {
  "teams": Array [
    Card {
      "count": 3,
      "label": "MOCK_TEAM",
      "tags": Object {
        "allocated": 1,
        "overdue": 0,
      },
      "type": "team",
      "value": 1,
    },
    Card {
      "count": 1,
      "label": "MOCK_TEAM",
      "tags": Object {
        "allocated": 1,
        "overdue": 0,
      },
      "type": "team",
      "value": 2,
    },
  ],
  "user": Array [
    Card {
      "count": 1,
      "label": "Cases",
      "tags": Object {
        "allocated": undefined,
        "overdue": false,
      },
      "type": undefined,
      "value": undefined,
    },
  ],
}
`;

exports[`Team Workstack Adapter should transform a stage array to a team workstack schema 1`] = `
Object {
  "allocateToTeamEndpoint": "/allocate/team",
  "allocateToUserEndpoint": "/allocate/user",
  "allocateToWorkstackEndpoint": "/unallocate/",
  "columns": Array [],
  "dashboard": Array [
    Card {
      "count": 4,
      "label": "MOCK_CASETYPE",
      "tags": Object {
        "allocated": 1,
        "overdue": 2,
      },
      "type": "workflow",
      "value": "DEFAULT",
    },
    Card {
      "count": 4,
      "label": "MOCK_CASETYPE",
      "tags": Object {
        "allocated": 2,
        "overdue": 2,
      },
      "type": "workflow",
      "value": "WCS",
    },
  ],
  "items": Array [
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234567/19",
      "caseType": "DEFAULT",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-01",
      "deadlineDisplay": "01/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234567/19",
      "caseType": "DEFAULT",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2020-12-11",
      "deadlineDisplay": "11/12/2020",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": false,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234567/19",
      "caseType": "DEFAULT",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2020-12-11",
      "deadlineDisplay": "11/12/2020",
      "stageType": "A",
      "stageTypeDisplay": "Closed",
      "stageTypeWithDueDateDisplay": "Closed",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234568/19",
      "caseType": "DEFAULT",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-02",
      "deadlineDisplay": "02/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234568/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2200-01-03",
      "deadlineDisplay": "03/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 2,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-04",
      "deadlineDisplay": "04/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": false,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "06-01-2200",
      "deadlineDisplay": null,
      "stageType": "A",
      "stageTypeDisplay": "Closed",
      "stageTypeWithDueDateDisplay": "Closed",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234570/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-05",
      "deadlineDisplay": "05/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
  ],
  "label": "MOCK_TEAM",
  "teamMembers": Array [],
}
`;

exports[`Team Workstack Adapter should transform a stage array to a team workstack schema ordered by systemCalculatedPriority 1`] = `
Object {
  "allocateToTeamEndpoint": "/allocate/team",
  "allocateToUserEndpoint": "/allocate/user",
  "allocateToWorkstackEndpoint": "/unallocate/",
  "columns": Array [],
  "dashboard": Array [
    Card {
      "count": 4,
      "label": "MOCK_CASETYPE",
      "tags": Object {
        "allocated": 2,
        "overdue": 2,
      },
      "type": "workflow",
      "value": "WCS",
    },
    Card {
      "count": 1,
      "label": "MOCK_CASETYPE",
      "tags": Object {
        "allocated": 1,
        "overdue": 1,
      },
      "type": "workflow",
      "value": "DEFAULT",
    },
  ],
  "items": Array [
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {
        "systemCalculatedPriority": 97,
      },
      "deadline": "1900-01-04",
      "deadlineDisplay": "04/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": false,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {
        "systemCalculatedPriority": 56.3,
      },
      "deadline": "06-01-2200",
      "deadlineDisplay": null,
      "stageType": "A",
      "stageTypeDisplay": "Closed",
      "stageTypeWithDueDateDisplay": "Closed",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234570/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {
        "systemCalculatedPriority": 43.9,
      },
      "deadline": "1900-01-05",
      "deadlineDisplay": "05/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234568/19",
      "caseType": "DEFAULT",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {
        "systemCalculatedPriority": 35,
      },
      "deadline": "1900-01-02",
      "deadlineDisplay": "02/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234568/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {
        "systemCalculatedPriority": 13.43,
      },
      "deadline": "2200-01-03",
      "deadlineDisplay": "03/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 2,
    },
  ],
  "label": "MOCK_TEAM",
  "teamMembers": Array [],
}
`;

exports[`User Workstack Adapter should transform a stage array to a user workstack schema 1`] = `
Object {
  "allocateToWorkstackEndpoint": "/unallocate/",
  "columns": Array [],
  "items": Array [
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseType": "DEFAULT",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "deadline": "2200-01-01",
      "deadlineDisplay": "01/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 1,
      "userUUID": 1,
    },
  ],
  "label": "My Cases",
}
`;

exports[`Workflow Workstack Adapter should transform a stage array to a workflow workstack schema 1`] = `
Object {
  "allocateToTeamEndpoint": "/allocate/team",
  "allocateToUserEndpoint": "/allocate/user",
  "allocateToWorkstackEndpoint": "/unallocate/",
  "columns": Array [],
  "dashboard": Array [
    Card {
      "count": 4,
      "label": "MOCK_STAGETYPE",
      "tags": Object {
        "allocated": 2,
        "overdue": 2,
      },
      "type": "stage",
      "value": "A",
    },
    Card {
      "count": 1,
      "label": "MOCK_STAGETYPE",
      "tags": Object {
        "allocated": 1,
        "overdue": 1,
      },
      "type": "stage",
      "value": "B",
    },
  ],
  "items": Array [
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234568/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2200-01-03",
      "deadlineDisplay": "03/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 2,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-04",
      "deadlineDisplay": "04/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2200-01-06",
      "deadlineDisplay": "06/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-07",
      "deadlineDisplay": "07/01/1900",
      "stageType": "B",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234570/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-05",
      "deadlineDisplay": "05/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
  ],
  "label": "MOCK_CASETYPE",
  "teamMembers": Array [],
}
`;

exports[`Workflow Workstack Adapter should transform a stage array to a workflow workstack schema 2`] = `
Object {
  "allocateToTeamEndpoint": "/allocate/team",
  "allocateToUserEndpoint": "/allocate/user",
  "allocateToWorkstackEndpoint": "/unallocate/",
  "columns": Array [],
  "items": Array [
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234568/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2200-01-03",
      "deadlineDisplay": "03/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 2,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-04",
      "deadlineDisplay": "04/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "assignedUserDisplay": "MOCK_USER",
      "caseReference": "A/1234569/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "2200-01-06",
      "deadlineDisplay": "06/01/2200",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": 1,
    },
    Object {
      "active": true,
      "assignedTeamDisplay": "MOCK_TEAM",
      "caseReference": "A/1234570/19",
      "caseType": "WCS",
      "caseTypeDisplayFull": "MOCK_CASETYPE",
      "data": Object {},
      "deadline": "1900-01-05",
      "deadlineDisplay": "05/01/1900",
      "stageType": "A",
      "stageTypeDisplay": "MOCK_STAGETYPE",
      "stageTypeWithDueDateDisplay": "MOCK_STAGETYPE",
      "teamUUID": 2,
      "userUUID": null,
    },
  ],
  "label": "MOCK_STAGETYPE",
  "teamMembers": Array [],
}
`;
